syntax = "proto3";
package namespace_Contacts;

// 引入any.proto文件
import "google/protobuf/any.proto";

message Address {
    string home_address = 1; // 家庭地址
    string unit_address = 2; // 单位地址
}

// 联系人
message PeopleInfo {
    string name = 1;
    int32 age = 2;
    // 嵌套
    message Phone {
        string number = 1;

        enum PhoneType {
            MP = 0; // 固定电话
            TEL = 1; // 移动电话
        }
        PhoneType type = 2;
    }
    repeated Phone phone = 3;
    google.protobuf.Any address = 4; // 保存地址信息

    // 不能在oneof里使用repeated，如果设置多个就会保留最后一个
    oneof other_contact{
        string qq = 5;
        string wechat = 6;
    }
}

// 通讯录
message Contacts {
    repeated PeopleInfo contacts = 1;
}